src/github/testgenerator.py

```diff
@@ -109,74 +137,170 @@ def detect_dependencies(self, file_path):
+                    model="gpt-4o",
+                    messages=[
+                        {"role": "system", "content": "Return only a Python list of strings"},
+                        {"role": "user", "content": f"{prompt}\n\nCode:\n{code}"}
+                    ],
+                    temperature=0
+                )
+                content = response.choices[0].message.content.strip()
+            else:  # anthropic
+                response = self.anthropic_client.messages.create(
+                    model="claude-3-5-sonnet-20241022",
+                    max_tokens=1000,
+                    messages=[
+                        {
+                            "role": "user",
+                            "content": "Return only a Python list of strings.\n\n" + f"{prompt}\n\nCode:\n{code}"
+                        }
+                    ]
+                )
+                content = response.content[0].text.strip()
+
+            if self.debug:
+                print(f"Debug - Raw AI response for {file_path}:")
+                print(content)
+            
+            # Clean up the response to ensure valid Python syntax
+            content = content.replace('```python', '').replace('```', '').strip()
+            if not (content.startswith('[') and content.endswith(']')):
+                if self.debug:
+                    print(f"Warning: Invalid list format received: {content}")
+                return []
+                
+            try:
+                deps = eval(content)
+                if not isinstance(deps, list):
+                    if self.debug:
+                        print(f"Warning: Non-list result evaluated: {deps}")
+                    return []
+                return deps
+            except SyntaxError as se:
+                if self.debug:
+                    print(f"Syntax error in AI response: {se}")
+                    print(f"Problematic content: {content}")
+                return []
+            
         except Exception as e:
-            print(f"Error detecting dependencies: {e}")
+            if self.debug:
+                print(f"Error detecting dependencies for {file_path}: {str(e)}")
+                print(f"File content: {code[:200]}...")  # Print first 200 chars of file
             return []
 
     def update_dependencies(self, services):
         pom_path = Path(self.local_path) / "pom.xml"
         if not pom_path.exists():
             return
 
-        prompt = f"""
-        Add these Testcontainers dependencies to pom.xml for services {services}.
-        Also include:
-        1. JUnit Jupiter dependency
-        2. Testcontainers JUnit Jupiter dependency
-        3. Testcontainers BOM
-        4. Spring Boot Test dependency
+        # Read existing pom.xml
+        pom_content = pom_path.read_text()
         
-        Current pom.xml content:
-        {pom_path.read_text()}
+        # Check which dependencies we need to add
```

Comment:
So far we can leave this implementation but moving forward we should update this part to a prompt. There are 100+ Testcontainers modules available, so here should be something more generic. For the demo purposes postgresql, mongodb, redis are good-enough.